vsphere/resource_vsphere_compute_cluster.go

```diff
@@ -1415,7 +1448,59 @@ func expandClusterConfigSpecEx(d *schema.ResourceData, version viapi.VSphereVers
 		obj.ProactiveDrsConfig = expandClusterProactiveDrsConfigInfo(d)
 	}
 
-	return obj
+	return obj, nil
+}
+
+func expandVsanHostConfig(d *schema.ResourceData, obj []types.VsanHostConfigInfo) []types.VsanHostConfigInfo {
```

Comment:
it feels strange to me to be taking in []types.VsanHostConfigInfo and also return the same type. Could you add a code comment (maybe even a few sentences in length) somewhere describing the rationale of this in general. I'm curious to now why we are performing the "joining" of the sets?

Reply:
Are we not able to just pass along whatever the user has specified?

Reply:
putting hosts into fault domain is not very straight forward, this should be done by inserting one field of `VsanHostConfig`, which content is:
```
type VsanHostConfigInfo struct {
	DynamicData

	Enabled         *bool                          `xml:"enabled"`
	HostSystem      *ManagedObjectReference        `xml:"hostSystem,omitempty"`
	ClusterInfo     *VsanHostConfigInfoClusterInfo `xml:"clusterInfo,omitempty"`
	StorageInfo     *VsanHostConfigInfoStorageInfo `xml:"storageInfo,omitempty"`
	NetworkInfo     *VsanHostConfigInfoNetworkInfo `xml:"networkInfo,omitempty"`
	FaultDomainInfo *VsanHostFaultDomainInfo       `xml:"faultDomainInfo,omitempty"`
	VsanEsaEnabled  *bool                          `xml:"vsanEsaEnabled"`
}
```
After the cluster created, there're other info tracked in this data structure, we should keep them and just update the `FaultDomainInfo`, that's why we pass and return same type. Because the changes we make is something like insertion. We'll explain this in comment.